### DISCUSSION TITLE
Unable to resolve "@storybook/addon-react-native-web/register"

### DISCUSSION THREAD

**Discussion by 7SilviaT** [storybookjs/storybook#24775] [Help] [answered]
I am working on a RN app and the task is implement storybook and visualize components in web and emulators(android/ios). Following the steps in this link: [https://levelup.gitconnected.com/introducing-react-native-web-storybook-3c56d48a6508](url) I have the error Unable to resolve "@storybook/addon-react-native-web/register" from ".storybook/storybook.requires.js" when run storybook in emulator ( displays in the web fine).

The package react-native-web is already installed, and yhe package json is:

`"scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web",
    "storybook-generate": "sb-rn-get-stories",
    "storybook-watch": "sb-rn-watcher",
    "storybook": "sb-rn-get-stories && cross-env STORYBOOK_ENABLED='true' expo start",
    "storybook:ios": "sb-rn-get-stories && cross-env STORYBOOK_ENABLED='true' expo start --ios",
    "storybook:android": "sb-rn-get-stories && cross-env STORYBOOK_ENABLED='true' expo start --android",
    "storybook:web": "start-storybook -p 6006 -c .storybook-web",
    "build-storybook": "build-storybook -c .storybook-web"
  },

![emulator-response](https://github.com/storybookjs/storybook/assets/97373107/f5aaab5c-0535-49d1-a4ef-d02b64fef3d1)
`
Is the script to run storybook in emulator correct?
When running the script sb-rn-stories the storybook.requires.js  is:

`/* do not change this file, it is auto generated by storybook. */

import {
  configure,
  addDecorator,
  addParameters,
  addArgsEnhancer,
  clearDecorators,
} from "@storybook/react-native";

global.STORIES = [
  {
    titlePrefix: "",
    directory: "./components",
    files: "*/.stories.?(ts|tsx|js|jsx)",
    importPathMatcher:
      "^\\.[\\\\/](?:components(?:\\/(?!\\.)(?:(?:(?!(?:^|\\/)\\.).)?)\\/|\\/|$)(?!\\.)(?=.)[^/]?\\.stories\\.(?:ts|tsx|js|jsx)?)$",
  },
];

import "@storybook/addon-ondevice-controls/register";
import "@storybook/addon-ondevice-actions/register";
import "@storybook/addon-react-native-web/register";

import { argsEnhancers } from "@storybook/addon-actions/dist/modern/preset/addArgs";

import { decorators, parameters } from "./preview";

if (decorators) {
  if (_DEV_) {
    // stops the warning from showing on every HMR
    require("react-native").LogBox.ignoreLogs([
      "`clearDecorators` is deprecated and will be removed in Storybook 7.0",
    ]);
  }
  // workaround for global decorators getting infinitely applied on HMR, see https://github.com/storybookjs/react-native/issues/185
  clearDecorators();
  decorators.forEach((decorator) => addDecorator(decorator));
}

if (parameters) {
  addParameters(parameters);
}

try {
  argsEnhancers.forEach((enhancer) => addArgsEnhancer(enhancer));
} catch {}

const getStories = () => {
  return {
    "./components/Button/Button.stories.tsx": require("../components/Button/Button.stories.tsx"),
    "./components/Link/Link.stories.tsx": require("../components/Link/Link.stories.tsx"),
    "./components/TextInput/TextInput.stories.tsx": require("../components/TextInput/TextInput.stories.tsx"),
  };
};

configure(getStories, module, false);`

Any help is appreciated,thanks in advance.

**Comment by dannyhw** [comment, marked as answer]
@7SilviaT you can't include addon-react-native-web in the react native addons since its a web specific addon. 

addon-react-native-web is used on a reactjs storybook to make stories load on the web.

**Reply by joevaugh4n** [reply]
Thanks for jumping in here, @dannyhw!

**Comment by nathanbedford** [comment]
@dannyhw I'm a bit confused - the [readme](https://github.com/storybookjs/addon-react-native-web/blob/main/README.md) indicates we should be adding this to the 'addon's property in main.js::

**.storybook/main.js**
```
module.exports = {
  addons: [/*existing addons,*/ '@storybook/addon-react-native-web'],
}
```

**Reply by dannyhw** [reply]
not for the ondevice storybook, it should be on the web storybook

**Reply by dannyhw** [reply]
usually you have a config for web and another for native

**Reply by nathanbedford** [reply]
Thanks for the clarification!
